core/domain/interaction_jobs_one_off.py

```diff
@@ -392,3 +393,34 @@ def get_invalid_values(value_type, value, choices):
     @staticmethod
     def reduce(key, values):
         yield (key, values)
+
+
+class LogicProofInteractionOneOffJob(jobs.BaseMapReduceOneOffJobManager):
+    """Job that produces a list of (exploaration, state) pairs that use the
+    logicProof interaction. This will be used to send manual emails.
+    This job can be deleted after logicProof is removed from the codebase.
+    """
+
+    @classmethod
+    def entity_classes_to_map_over(cls):
+        return [exp_models.ExplorationModel]
+
+    @staticmethod
+    def map(item):
+        if item.deleted:
+            return
+        exploration = exp_fetchers.get_exploration_from_model(item)
+        for _, state in exploration.states.items():
+            if state.interaction.id == 'LogicProof':
+                owner_ids = rights_manager.get_exploration_rights(
+                    item.id).owner_ids
+                # Always yield for all case ?
+                if owner_ids:
```

Comment:
Please remove, this is redundant since if `owner_ids` is empty the loop's code would not be run anyway.

Reply:
got it. Done!